netlogo-gui/src/main/ide/ShowUsageBox.scala

```diff
@@ -0,0 +1,126 @@
+// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo
+
+package org.nlogo.ide
+
+import java.awt.Color
+import java.awt.event._
+import javax.swing.table.{DefaultTableCellRenderer, DefaultTableModel}
+import javax.swing.{JDialog, JScrollPane, JTable}
+
+import org.nlogo.core.{Femto, Token, TokenizerInterface}
+import org.nlogo.editor.EditorArea
+
+class ShowUsageBox {
+  val usageBox = new JDialog()
+  val dataModel = new DefaultTableModel(){
+    override def isCellEditable(row: Int, column: Int): Boolean = false
+    override def getColumnClass(columnIndex: Int): Class[_] = {
+      if(columnIndex == 0) classOf[Token] else super.getColumnClass(columnIndex)
+    }
+  }
+  val usageTable = new JTable(dataModel)
+  val scrollPane = new JScrollPane(usageTable)
+  var editorArea: EditorArea = null
+  usageBox.add(scrollPane)
+  usageBox.setUndecorated(true)
+  usageTable.setDefaultRenderer(classOf[Token], new LineNumberRenderer())
+
+  def init(editorArea: EditorArea): Unit = {
+    if(this.editorArea != null){
+      return
+    }
+    this.editorArea = editorArea
+    usageBox.addWindowFocusListener(new WindowFocusListener {
+      override def windowLostFocus(e: WindowEvent): Unit = {
+        usageBox.setVisible(false)
+        dataModel.synchronized(dataModel.setRowCount(0))
+      }
+
+      override def windowGainedFocus(e: WindowEvent): Unit = {}
+    })
+    dataModel.setColumnCount(2)
+    usageTable.setSelectionMode(javax.swing.ListSelectionModel.SINGLE_SELECTION)
+    usageTable.getColumnModel.getColumn(0).setMaxWidth(40)
+    usageTable.getColumnModel.setColumnSelectionAllowed(false)
+    usageTable.getTableHeader().setReorderingAllowed(false)
+    usageTable.setTableHeader(null)
+    usageTable.setRowMargin(3)
+    usageTable.setRowHeight(20)
+//    usageTable.setBackground(Color.WHITE)
+    usageTable.getColumnModel.getColumn(1).setMinWidth(200)
+    usageTable.getColumnModel.getColumn(0).setMinWidth(40)
+    usageTable.setFont(editorArea.getFont)
+
+    usageTable.addMouseListener(new MouseListener(){
+      override def mouseClicked(e: MouseEvent): Unit = {}
+
+      override def mouseExited(e: MouseEvent): Unit = {}
+
+      override def mouseEntered(e: MouseEvent): Unit = {}
+
+      override def mousePressed(e: MouseEvent): Unit = {}
+
+      override def mouseReleased(e: MouseEvent): Unit = {
+        usageBox.setVisible(false)
+        val token = usageTable.getValueAt(usageTable.getSelectedRow, 0).asInstanceOf[Token]
+        dataModel.synchronized(dataModel.setRowCount(0))
+        println(token.start + " " + token)
```

Comment:
I know this is here for debugging, just commenting to make sure we remember to remove it before we merge in.